src/deepsparse/transformers/pipelines/text_generation.py

```diff
@@ -601,6 +610,36 @@ def _stream_engine_outputs(
                 **schema_kwargs,
             )
 
+    def _generate_streamed_text_from_last_n_tokens(
+        self, generated_tokens: numpy.ndarray, prompts: str, n: int = 4
+    ) -> str:
+        """
+        An auxiliary method that helps to properly generate the streamed text.
+        Some models like llama2 and mistral are using LlamaTokenizer which is
+        based on SentencePiece tokenizer. This specific tokenizer doesn't seem
+        to output appropriate prefix spaces when decoding token by token.
+        One can make it work if the previously generated tokens are included.
+        This allows the tokenizer to figure out that the appropriate spaces
+        from last n consecutive tokens.
+
+        :param generated_tokens: the generated tokens from the engine
+        :param prompts: the input prompts (needed to inititalize
+            the list of last n tokens)
+        :param n: the number of last tokens to use for decoding
+        :return: the generated string
+        """
+        if not hasattr(self, "last_n_tokens"):
+            self.last_n_tokens = self.tokenizer(prompts).input_ids[-n:]
+        string_from_n_tokens = self.tokenizer.decode(
+            self.last_n_tokens, skip_special_tokens=True
+        )
+        self.last_n_tokens.append(generated_tokens[0])
+        string_from_n_plus_1_tokens = self.tokenizer.decode(
+            self.last_n_tokens, skip_special_tokens=True
+        )
+        self.last_n_tokens.pop(0)
+        return string_from_n_plus_1_tokens[len(string_from_n_tokens) :]
```

Comment:
slices are inneficient, is there something else we can do

Reply:
Should not be too much of an overhead (per: https://stackoverflow.com/questions/16891340/remove-a-prefix-from-a-string)